File: truss-compiler/src/test/java/truss/compiler/TestCase.java
package truss.compiler;

public class TestCase {
    private final StringBuilder sb = new StringBuilder();
    private final String name;
    private final String fileName;
    private String code;

    public TestCase(String name, String fileName) {
        this.name = name;
        this.fileName = fileName;
    }

    public void beginExpected() {
        code = sb.toString();
        sb.setLength(0);
    }

    public void append(String line) {
        sb.append(line);
        sb.append("\n");
    }

    public void test(TestCaseCallback callback) throws Exception {
        String expected;
        if (code == null) {
            code = sb.toString();
            expected = code;
        } else {
            expected = sb.toString();
        }

        callback.test(name, fileName, code.trim(), expected.trim());
    }
}

File: Truss.Compiler.Test/TestCase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Test {
    public class TestCase {
        private readonly StringBuilder _sb = new StringBuilder();
        private readonly String _name;
        private readonly String _fileName;
        private String _code;

        public TestCase(String name, String fileName) {
            _name = name;
            _fileName = fileName;
        }

        public void BeginExpected() {
            _code = _sb.ToString();
            _sb.Clear();
        }

        public void Append(String line) {
            _sb.AppendLine(line);
        }

        public void Test(ITestCaseCallback callback) {
            String expected;
            if (_code == null) {
                _code = _sb.ToString();
                expected = _code;
            } else {
                expected = _sb.ToString();
            }

            callback.Test(_name, _fileName, _code.Trim(), expected.Trim());
        }
    }
}
